reusable-workflows/build.yml

```diff
@@ -82,6 +80,8 @@ jobs:
       - name: Build
         if: steps.s3-cache.outputs.processed == 'false'
         run: ${{ inputs.build_cmd }}
+        env:
+          DD_API_KEY: ${{ github.ref == 'refs/heads/main' && secrets.DD_API_KEY || null }}
```

Comment:
Not sure and potentially nitpicky comment, is it clear from the code itself why we're only providing the API_KEY on main?
I'm thinking whether we want to 
1. add a comment why there's this check here, or
2. move the if statement whether we want to use the build plugin in a separate env, for example `DD_REPORT_BUILD_SIZE: ${{ github.ref == 'refs/heads/main'}}`

Reply:
I think the extra variable makes it more explicit  rather than implicit. Some advantages I can think of
1. Devs won't need to read a comment to understand the code - the code will be more self-explanatory (personal preference)

2. currently, if we stop sending down the `DD_API_KEY`, we will stop sending the metrics to Datadog without even knowing it. If we have an explicit variable, we can throw error while setting up the build plugin on missing `DD_API_KEY` variable.

Anyway, this indeed is a nitpick and adding just a comment is fine by me as well 😄

Reply:
Added here https://github.com/pleo-io/spa-tools/pull/179/commits/0536a7a1f3a0886a26ad553dc3e2303f66358cad